unittests/savanna_cluster.hpp

```diff
@@ -10,6 +10,43 @@
 
 #include "snapshot_suites.hpp"
 
+// ---------------------------------------------------------------------------
+// An object which assigns a value to a variable in its constructor, and resets
+// to its previous value in its destructor
+// ---------------------------------------------------------------------------
+template <class T>
+class scoped_set_value {
```

Comment:
Consider moving to `fc` for reuse.

Reply:
Yes, sounds good. I moved it to `fc/scoped_exit.hpp` as I don't think we need a file for every class but let me know if you think it is better to create its own file.

Reply:
I  feel like renaming `scoped_exit.hpp` to `scoped_utils.hpp`. What do you think?

Reply:
I think it is fine as is.